performance-tests/bench/node_controller/main.cpp

```diff
@@ -187,18 +189,27 @@ class SpawnedProcess {
       }
       ss << command << std::flush;
     }
-    const std::string command = ss.str();
-    std::cerr << command + "\n" << std::flush;
-    proc_opts->command_line("%s", command.c_str());
+    command_line_ = ss.str();
+    proc_opts->command_line("%s", command_line_.c_str());
     return proc_opts;
   }
 
   void set_pid(pid_t pid) noexcept
   {
+    start_time_ = Builder::get_sys_time();
     pid_ = pid;
     running_ = true;
   }
 
+  void log_process_creation()
+  {
+    std::stringstream ss;
+    ss << Bench::iso8601() << ": Node " << node_id_ << " (" << node_name_ << ") starting process " << pid_ << ":\n";
+    ss << " - original config: " << original_config_name_ << "\n";
+    ss << " - command line: " << command_line_ << "\n\n" << std::flush;
+    std::cerr << ss.str() << std::flush;
```

Comment:
Why put everything into a string stream first and not into `cerr` directly? Also I don't think string stream has anything to flush unlike for file I/O.

Reply:
Since cerr is a shared resource, putting it all into a stringstream first reduces the number of operations on cerr and prevents having the lines from being split up when multiple threads and/or processes are writing at the the same time.

Reply:
Alright, that makes sense. What about the flush?